src/java.base/share/classes/java/lang/invoke/BoundMethodHandle.java

```diff
@@ -360,6 +360,15 @@ SpeciesData extendWith(byte typeNum) {
             extensions[typeNum] = sd;
             return sd;
         }
+
+        SpeciesData extendWith(BasicType basicType) {
+            int typeNum = basicType.ordinal();
+            SpeciesData sd = extensions[typeNum];
+            if (sd != null)  return sd;
+            sd = SPECIALIZER.findSpecies(key() + basicType.basicTypeChar());
```

Comment:
As soon as this is `java.base` I suggest to replace `key() + basicType.basicTypeChar()` with `key().concat(String.valueOf(basicType.basicTypeChar()))` to prevent `StringBuilder` allocation and `append()` chaining.

Reply:
There are only 5 elements in `extensions` arrays for 5 `BasicType`.  I don't think this overhead is significant.